Utilities/XrdAdaptor/README.md

```diff
@@ -0,0 +1,49 @@
+# XrdAdaptor
+
+## Introduction
+
+The `XrdAdaptor` package is the CMSSW implementation of CMS' AAA infrastructure. The main features on top of the stock XRootD client library are
+* Recovery from some errors via re-tries
+* Use of multiple XRootD sources (described further [here](doc/multisource_algorithm_design.md))
+
+## Short description of components
+
+### `ClientRequest`
+
+The `ClientRequest` implements `XrdCl::ResponseHandler`, and represents a single read request(?).
+
+### `QualityMetric` etc
+
+?
+
+### `RequestManager`
+
+The `RequestManager` containes the actual implementation of the retries and the multi-source algorithm. There is one `RequestManager` object for one PFN, and it contains one or more `Source` objects.
+
+#### `RequestManager::OpenHandler`
+
+The `OpenHandler` implements XRootD's `XrdCl::ResponseHandler` in an asynchronous way(?). An instance is created in `RequestManager::initialize()`, and used when additional Sources are opened, either as part of the multi-source comparisons (`RequestManager::checkSourcesImpl()`) or read error recovery (`RequestManager::requestFailure()`).
+
+### `Source`
+
+The `Source` represents a connection to one storage server. There can be more than one `Source` for one PFN. 
+
+### `SyncHostResponseHandler`
+
+The `SyncHostResponseHandler` implements XRootD's `XrdCl::ResponseHandler` in a synchronous way(?). It is used in `RequestManager::initialize()` for the initial file open.
+
+### `XrdFile`
+
+The `XrdFile` implements `edm::storage::Storage` (see [Utilities/StorageFactory/README.md](../../Utilities/StorageFactory/README.md). In CMS' terminology it represents one Phsyical File Name (PFN), and acts as a glue between the `edm::storage::Storage` API and `RequestManager`.
+
+### `XrdStatistics`
+
```

Comment:
What do you wish to add here, is `XrdStatistics` more than what it described here, https://github.com/cms-sw/cmssw/blob/275ceb6c19b3b860a5d691be1ab15e47949266c7/Utilities/XrdAdaptor/interface/XrdStatistics.h#L20-L23

Reply:
I'd like to have some description of how the `XrdStatistics` is being used between `XrdStatisticsService` and the rest of `XrdAdaptor`.

Description of the statistics would also be nice.

Reply:
Here's what I would suggest:

> `XrdStatistics` provides per-"site" counters (bytes read, total time), providing the statistics service
> with a viewpoint of how individual sites are performing for a given client.
> The intent is to provide more visibility into the performance encountered by grouping I/O operations
> into a per-site basis, under the theory that performance within a site is similar but could differ between
> two different sites.